arrabiata/src/interpreter.rs

```diff
@@ -0,0 +1,962 @@
+//! This module contains the implementation of the IVC scheme in addition to
+//! running an arbitrary function that can use up to [crate::NUMBER_OF_COLUMNS]
+//! columns.
+//! At the moment, all constraints must be of maximum degree
+//! [crate::MAX_DEGREE], but it might change in the future.
+//!
+//! The implementation relies on a representation of the circuit as a 2D array
+//! of "data points" the interpreter can use.
+//!
+//! An interpreter defines what a "position" is in the circuit and allow to
+//! perform operations using these positions.
+//! Some of these positions will be considered as public inputs and might be
+//! fixed at setup time while making a proof, when other will be considered as
```

Comment:
public inputs are not fixed at setup time
Note that in some use cases some PI are also computed by the prover
eg. crafting a proof that a transition from an intial state to a resulting state requires the prover to compute the resulting state, although it is marked as a PI

Reply:
The sentence is not phrased correctly. The "and might be fixed at setup time" are not the public inputs, but for "positions".